lectures/stack/stack_tests.hpp

```diff
@@ -90,6 +92,57 @@ TEST_CASE_TEMPLATE("Проверка дали промяна на присвое
     }
 }
 
+TEST_CASE_TEMPLATE("Проверка за самоприсвояване чрез копиране", SomeStack, STACKS) {
+    SomeStack s;
+    s.push(1);
+    s.push(2);
+
+    s = s;
+
+    CHECK(s.pop() == 2);
+    CHECK(s.pop() == 1);
+    CHECK(s.empty());
+}
+
+TEST_CASE_TEMPLATE("Проверка за самоприсвояване чрез преместване", SomeStack, MOVEABLE_STACKS) {
+    SomeStack s;
+    s.push(1);
+    s.push(2);
+
+    s = std::move(s);
+
+    CHECK(s.pop() == 2);
+    CHECK(s.pop() == 1);
+    CHECK(s.empty());
+}
+
+TEST_CASE_TEMPLATE("Проверка дали преместването нa стек на мястото на друг стек e успешно", SomeStack, MOVEABLE_STACKS) {
+    SomeStack dummy;
+    dummy.push(1);
+    dummy.push(2);
+
+    SUBCASE("Проверка на Move constructor") {
+        SomeStack s(std::move(dummy));
+        CHECK(s.pop() == 2);
+        CHECK(s.pop() == 1);
+        CHECK(s.empty());
+        CHECK_THROWS(dummy.peek());
```

Comment:
Тук може да е просто
```suggestion
        CHECK(dummy.empty());
```
долу по същия начин